### DISCUSSION TITLE
How to add email to a user sign in with apple oAuth provider

### DISCUSSION THREAD

**Discussion by akshaykumarp12** [pocketbase/pocketbase#5068] [Q&A]
When a user is sign in with apple, there is no email registerd in the user collection. When trying to change email in user collection with email change request , a password is prompted when clicked on the verification link sent to that email. for wich the user doesnt added any password already.
![bb](https://github.com/pocketbase/pocketbase/assets/167839699/3acdd33a-1e9a-487c-a2dd-50dc9ab28227)

**Comment by ganigeorgiev** [comment]
Email change currently always require a password for confirmation from regular users.

At the moment the email set/change of an OAuth2 only created account is not clearly defined but one workaround could be to explicitly set a client-side generated password during the OAuth2 auth and treat it as a "security code", something like:

```dart
final password = "..." // some random generated password

// assuming that you use the Dart SDK
await pb.collection("users").authWithOAuth2(
  'google',
  (url) async {
    await launchUrl(url);
  },
  createData: {
    'password': password,
    'passwordConfirm': password
  },
);

// on success show the password to the user and present it as a "security code" that they will need to store somewhere safe
```

The above flow is certainly not ideal and eventually may change in the future as part of [this task](https://github.com/orgs/pocketbase/projects/2/views/1?filterQuery=change+email&pane=issue&itemId=22401932).

About why the Apple provider may not return an email - see https://github.com/pocketbase/pocketbase/discussions/3814. The Apple OAuth2 provider was implemented based on only my understanding of their docs since I don't have Apple developer account to test it. I suspect it will require adding another Apple only case for handling POST redirects but as mentioned in the linked discussion I'm not planning investing time on this at the moment. I'll accept PR If the required changes are not so drastic.

**Reply by ganigeorgiev** [reply]
_Side-note: I've pushed a small change in v0.22.14-rc that should allow Apple to return the user email but needs help with the testing. If you have the time please check https://github.com/pocketbase/pocketbase/discussions/5074._
